### PR TITLE
HOTFIX commented remove colomns category place problem

### PR DESCRIPTION
**Pull request by VKislov**

### CODE REVIEW

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 1]
```diff
@@ -8,82 +8,82 @@ exports.up = async (knex) => {
...
 --
-ALTER TABLE "TicketHistoryRecord" DROP COLUMN "problemClassifier" CASCADE;
+--ALTER TABLE "TicketHistoryRecord" DROP COLUMN "problemClassifier" CASCADE;
 --
 -- Add field classifierRuleDisplayNameFrom to ticketchange
 --
 ALTER TABLE "TicketChange" ADD COLUMN "classifierRuleDisplayNameFrom" text NULL;
 --
--- Add field classifierRuleDisplayNameTo to ticketchange
+Add field classifierRuleDisplayNameTo to ticketchange
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:78]: ?

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 2]
```diff
@@ -8,82 +8,82 @@ exports.up = async (knex) => {
...
 --
 ALTER TABLE "TicketChange" ADD COLUMN "classifierRuleDisplayNameTo" text NULL;
 --
 -- Add field classifierRuleIdFrom to ticketchange
 --
 ALTER TABLE "TicketChange" ADD COLUMN "classifierRuleIdFrom" uuid NULL;
 --
--- Add field classifierRuleIdTo to ticketchange
+Add field classifierRuleIdTo to ticketchange
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:86]: ?

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 3]
```diff
@@ -113,67 +113,67 @@ ALTER TABLE "TicketChange" DROP COLUMN "classifierRuleDisplayNameFrom" CASCADE;
 --
 -- Remove field problemClassifier from tickethistoryrecord
 --
-ALTER TABLE "TicketHistoryRecord" ADD COLUMN "problemClassifier" uuid NULL;
+--ALTER TABLE "TicketHistoryRecord" ADD COLUMN "problemClassifier" uuid NULL;
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:116]: `ALTER TABLE table_name ADD COLUMN IF NOT EXISTS column_name INTEGER;`

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 4]
```diff
@@ -113,67 +113,67 @@ ALTER TABLE "TicketChange" DROP COLUMN "classifierRuleDisplayNameFrom" CASCADE;
...
+ALTER TABLE "Ticket" ADD COLUMN IF NOT EXISTS "problemClassifier" uuid NULL CONSTRAINT "Ticket_problemClassifier_f1bfcd39_fk_TicketProblemClassifier_id" REFERENCES "TicketProblemClassifier"("id") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "Ticket_problemClassifier_f1bfcd39_fk_TicketProblemClassifier_id" IMMEDIATE;
 --
 -- Remove field placeClassifier from ticket
 --
-ALTER TABLE "Ticket" ADD COLUMN "placeClassifier" uuid NULL CONSTRAINT "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" REFERENCES "TicketPlaceClassifier"("id") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" IMMEDIATE;
+ALTER TABLE "Ticket" ADD COLUMN IF NOT EXISTS "placeClassifier" uuid NULL CONSTRAINT "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" REFERENCES "TicketPlaceClassifier"("id") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" IMMEDIATE;
 CREATE INDEX "Ticket_problemClassifier_f1bfcd39" ON "Ticket" ("problemClassifier");
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:177]: if not exists

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 5]
```diff
@@ -113,67 +113,67 @@ ALTER TABLE "TicketChange" DROP COLUMN "classifierRuleDisplayNameFrom" CASCADE;
...
 CREATE INDEX "Ticket_placeClassifier_01e3ec3a" ON "Ticket" ("placeClassifier");
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:178]: if not exists

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 6]
```diff
@@ -8,68 +8,68 @@ exports.up = async (knex) => {
 -- Remove field placeClassifier from ticket
 --
 SET CONSTRAINTS "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" IMMEDIATE; ALTER TABLE "Ticket" DROP CONSTRAINT "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id";
```
Comment (AntonAL) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:10]: If we want to save `Ticket.placeClassifier` column, maybe we don't need to drop a constraint. It does not disturbs us, AFAIK.

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 7]
```diff
@@ -113,67 +113,67 @@ ALTER TABLE "TicketChange" DROP COLUMN "classifierRuleDisplayNameFrom" CASCADE;
...
+ALTER TABLE "Ticket" ADD COLUMN IF NOT EXISTS "problemClassifier" uuid NULL CONSTRAINT "Ticket_problemClassifier_f1bfcd39_fk_TicketProblemClassifier_id" REFERENCES "TicketProblemClassifier"("id") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "Ticket_problemClassifier_f1bfcd39_fk_TicketProblemClassifier_id" IMMEDIATE;
 --
 -- Remove field placeClassifier from ticket
 --
-ALTER TABLE "Ticket" ADD COLUMN "placeClassifier" uuid NULL CONSTRAINT "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" REFERENCES "TicketPlaceClassifier"("id") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" IMMEDIATE;
+ALTER TABLE "Ticket" ADD COLUMN IF NOT EXISTS "placeClassifier" uuid NULL CONSTRAINT "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" REFERENCES "TicketPlaceClassifier"("id") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" IMMEDIATE;
```
Comment (AntonAL) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:176]: First command in this line throws an error.
Check whether you already have this constraint locally.

**Comment by AntonAL** [comment]
A `Ticket` schema has a comment `// Todo(zuch): add placeClassifier, categoryClassifier and classifierRule`, make sure that this comment is actual.

**Comment by AntonAL** [comment]
Before merging, let's take a look at it together.
Invite me in a meet.

**Review by pahaz — CHANGES_REQUESTED** [review]
Looks to strange

File: apps/condo/migrations/20220801183319-0160_rename_classifierruledisplaynamefrom_ticketchange_classifierdisplaynamefrom_and_more.js [hunk 8]
```diff
@@ -27,31 +27,32 @@ ALTER TABLE "TicketHistoryRecord" RENAME COLUMN "classifierRule" TO "classifier"
...
-ALTER TABLE "Ticket" DROP COLUMN "classifierRule" CASCADE;
+--SET CONSTRAINTS "Ticket_classifierRule_f7a8dbc1_fk_TicketClassifierRule_id" IMMEDIATE; ALTER TABLE "Ticket" DROP CONSTRAINT "Ticket_classifierRule_f7a8dbc1_fk_TicketClassifierRule_id";
+ALTER TABLE "Ticket" RENAME COLUMN "classifierRule" TO "classifier";
+ALTER TABLE "Ticket" RENAME CONSTRAINT "Ticket_classifierRule_f7a8dbc1_fk_TicketClassifierRule_id" TO "Ticket_classifier_7a786997_fk_TicketClassifierRule_id";
 --
 -- Add field classifier to ticket
 --
-ALTER TABLE "Ticket" ADD COLUMN "classifier" uuid NULL CONSTRAINT "Ticket_classifier_7a786997_fk_TicketClassifierRule_id" REFERENCES "TicketClassifierRule"("id") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "Ticket_classifier_7a786997_fk_TicketClassifierRule_id" IMMEDIATE;
+--ALTER TABLE "Ticket" ADD COLUMN IF NOT EXISTS "classifier" uuid NULL CONSTRAINT "Ticket_classifier_7a786997_fk_TicketClassifierRule_id" REFERENCES "TicketClassifierRule"("id") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "Ticket_classifier_7a786997_fk_TicketClassifierRule_id" IMMEDIATE;
```
Comment (VKislov) [inline apps/condo/migrations/20220801183319-0160_rename_classifierruledisplaynamefrom_ticketchange_classifierdisplaynamefrom_and_more.js:36]: need to add constrain

**Comment by AntonAL** [comment]
Not actual.
All the changes was merged in PR https://github.com/open-condo-software/condo/pull/2074/files

**Pull request closed without merging**